### PR TITLE
Update SPOSet WFopt related APIs

### PR DESCRIPTION
**Pull request by ye-luo**
## Proposed changes
Once TrialWF is built, it calls checkInVariables to query variational parameters under the TrialWF tree regardless of if running WFOpt or not. This the desired behavior.
Consequentially, checkInVariables cannot be implemented as a feature blocker. Update SPOSet and its derived classes.

## What type(s) of changes does this code introduce?
- Bugfix

### Does this introduce a breaking change?
- No

## What systems has this change been tested on?
epyc-server

## Checklist
- Yes. This PR is up to date with current the current state of 'develop'

### CODE REVIEW

**Review by PDoakORNL — COMMENTED** [review]
The obvious question is does something else need to be added to block this combination or was this just a message for developers.

**Comment by ye-luo** [comment]
There was a switch optimize at the determinant level and spo level. When there is a mismatch, need an error.
I'm moving away from this way because the determinant level doesn't include any parameter and should not own a switch.
Then there is no check needed since everything is decided by the spo. Simple.

**Review by PDoakORNL — DISMISSED** [review]
LGTM

**Comment by markdewing** [comment]
It looks like this PR also removes an unused feature for projecting the derivatives - multiplyDerivsByOrb and the "project" arguments.

File: src/QMCWaveFunctions/SPOSet.h [hunk 1]
```diff
@@ -117,10 +117,14 @@ class SPOSet : public QMCTraits
     APP_ABORT(o.str());
   }
   /// reset parameters to the values from optimizer
-  virtual void resetParameters(const opt_variables_type& optVariables) = 0;
+  virtual void resetParameters(const opt_variables_type& optVariables) {}
 
-  /// check in/out parameters to the global list of parameters used by the optimizer
+  /** check in parameters to the global list of parameters used by the optimizer.
+   * This is a query function and should never be implemented as a feature blocker.
```
Comment (markdewing) [inline src/QMCWaveFunctions/SPOSet.h:123]: I don't quite understand what's happening by looking at this comment and just the PR.  `resetParameters` is the function that changed from needing to be implemented by all the derived classes to one with a default implementation. `checkInVariables` already has a default implementation.  So this comment references some change that isn't in this PR?

Comment (ye-luo) [reply src/QMCWaveFunctions/SPOSet.h:123]: [LCAOrbitalSet.h](https://github.com/QMCPACK/qmcpack/pull/4136/files#diff-3008c4a5fdfa852f307be230c2d1e896b22871f8a9bf40fc2cdfd29be81da657)
I removed the blocker. So I added this comment to explain why.
I don't see a good reason to require implementing empty reset parameters.
There is a bigger picture I'm moving towards. https://github.com/ye-luo/qmcpack/wiki/Optimization-API-changes
It is still quite preliminary.

Comment (markdewing) [reply src/QMCWaveFunctions/SPOSet.h:123]: I missed the removal in LCAOrbitalSet. D'oh.

**Review by PDoakORNL — APPROVED** [review]
LGTM

**Comment by ye-luo** [comment]
Test this please

**Pull request merged**